test/algorithms/simulation/sidb/exhaustive_ground_state_simulation.cpp

```diff
@@ -21,57 +23,97 @@ TEMPLATE_TEST_CASE(
 
-        for (const auto& it : exgs_stats.valid_lyts)
-        {
-            CHECK(!it.charge_exists(sidb_charge_state::POSITIVE));
-        }
+        CHECK(exgs_stats.valid_lyts.size() == 1);
+        CHECK(exgs_stats.valid_lyts[0].get_charge_state_by_index(0) == sidb_charge_state::NEGATIVE);
     }
 
-    SECTION("simulation with one SiDBs placed")
+    SECTION("simulation with no SiDB placed")
     {
         TestType lyt{{20, 10}};
-        lyt.assign_cell_type({1, 3, 0}, TestType::cell_type::NORMAL);
 
         exgs_stats<TestType>             exgs_stats{};
         const sidb_simulation_parameters params{2, -0.32};
         exhaustive_ground_state_simulation<TestType>(lyt, params, &exgs_stats);
 
-        CHECK(exgs_stats.valid_lyts.size() == 1);
-        CHECK(exgs_stats.valid_lyts[0].get_charge_state_by_index(0) == sidb_charge_state::NEGATIVE);
+        CHECK(exgs_stats.valid_lyts.size() == 0);  // empty layout is saved.
     }
 
-    SECTION("simulation with no SiDBs placed")
+    SECTION("simulation of wire with one perturber and two SiDB pairs")
     {
         TestType lyt{{20, 10}};
 
+        lyt.assign_cell_type({0, 0, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({5, 0, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({7, 0, 0}, TestType::cell_type::NORMAL);
+
+        lyt.assign_cell_type({11, 0, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({13, 0, 0}, TestType::cell_type::NORMAL);
+
+        lyt.assign_cell_type({17, 0, 0}, TestType::cell_type::NORMAL);
+        lyt.assign_cell_type({19, 0, 0}, TestType::cell_type::NORMAL);
+
         exgs_stats<TestType>             exgs_stats{};
         const sidb_simulation_parameters params{2, -0.32};
         exhaustive_ground_state_simulation<TestType>(lyt, params, &exgs_stats);
+        auto size_before = exgs_stats.valid_lyts.size();
+        exhaustive_ground_state_simulation<TestType>(lyt, params, &exgs_stats);
+        auto size_after = exgs_stats.valid_lyts.size();
+        CHECK(size_before == 1);
+        CHECK(size_after == 1);
+
+        REQUIRE(!exgs_stats.valid_lyts.empty());
+        const auto& charge_lyt_first = exgs_stats.valid_lyts.front();
+
+        CHECK(charge_lyt_first.get_charge_state({0, 0, 0}) == sidb_charge_state::NEGATIVE);
+        CHECK(charge_lyt_first.get_charge_state({5, 0, 0}) == sidb_charge_state::NEUTRAL);
+        CHECK(charge_lyt_first.get_charge_state({7, 0, 0}) == sidb_charge_state::NEGATIVE);
+        CHECK(charge_lyt_first.get_charge_state({11, 0, 0}) == sidb_charge_state::NEUTRAL);
+        CHECK(charge_lyt_first.get_charge_state({13, 0, 0}) == sidb_charge_state::NEGATIVE);
+        CHECK(charge_lyt_first.get_charge_state({17, 0, 0}) == sidb_charge_state::NEUTRAL);
+        CHECK(charge_lyt_first.get_charge_state({19, 0, 0}) == sidb_charge_state::NEGATIVE);
+        CHECK_THAT(charge_lyt_first.get_system_energy(), Catch::Matchers::WithinAbs(0.24602741408, 0.00000001));
```

Comment:
Again, super minor: don't we have a constant for that small value? `POP_STABILITY_ERR` or something? Applies to both files.